lib/plotutil.py

```diff
@@ -68,12 +68,13 @@ def plot(func, xpoints, color_name, xlabel, ylabel, theme, gui, line_style, disc
         if not os.path.exists('.temp/'):
             os.mkdir('.temp/')
         plt.savefig(".temp/generated_plot.png")
+        plt.savefig(file_path)
```

Comment:
@aditodkar here the problem is, the plot will get saved only when user is using GUI mode because you have put the savefig(filepath) function call in the else part. The savefig(filepath) function call should be independent and outside the ```if not gui``` block
That is you need a separate block towards the end of the function:
```
if file_path != "":
  plt.savefig(file_path)
 # please put  a log statement here
```